espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/helpers/ViewFinder.java

```diff
@@ -222,6 +236,26 @@ private static boolean canScrollToViewWithContentDescription(@Nullable final Vie
         return true;
     }
 
+    private static List<View> getViewsFromDataInteraction(
+            @Nullable View root, DataInteraction dataInteraction
+    ) {
+        // Look up the view hierarchy to find the closest ancestor AdapterView
+        View ancestorAdapter = root;
+        while (ancestorAdapter != null && !(ancestorAdapter instanceof AdapterView)) {
+            ancestorAdapter = (View) ancestorAdapter.getParent();
+        }
+        if (ancestorAdapter != null) {
```

Comment:
let

Reply:
This is still Java. I'm going to convert these classes later.